File: fuzzbuzz.pl
#!/usr/local/bin/perl

print "Give me your end-number ";
$line = <STDIN>;
if ($line <= 0) { 
	print "You're crazy guy";
} 
else {
	$count = 0;
	while ($count < $line) {
		if ($count % 15 == 0) {
			print "FuzzBuzz\n";
		}
		elsif ($count % 5 == 0) {
			print "Fuzz\n";
		}
		elsif ($count % 3 == 0) {
			print "Buzz\n";
		}
		else{
			print "$count\n";
		}
	$count += 1;
	}
}

File: fuzzbuzz.hs
-- fuzzBuzz :: Int -> String
-- fuzzBuzz x
-- 	| (x `mod` 15 == 0) = "FuzzBuzz"
-- 	| (x `mod` 5 == 0) = "Fuzz"
-- 	| (x `mod` 3 == 0) = "Buzz"
-- 	| otherwise = show x

-- [fuzzBuzz x | x <- [1..20]]

fuzzBuzz2 :: Int -> [String]
fuzzBuzz2 xs = [fuzzBuzz x | x <- [1..xs]]
	where fuzzBuzz x
		| (x `mod` 15 == 0) = "FuzzBuzz"
		| (x `mod` 5 == 0) = "Fuzz"
		| (x `mod` 3 == 0) = "Buzz"
		| otherwise = show x